### DISCUSSION TITLE
How to do horizontal scaling?

### DISCUSSION THREAD

**Discussion by Richard87** [JasperFx/marten#2558] [Ideas]
Hi!

I am looking into Marten and was thinking about the scaling story.

I have read about running selected projections, but I'm not sure if they continue to run and wait for events, or if the finish after completing?

What I would love to see is
- **Running all instances with in-memory projections, with the Async Daemon watching the event-store for events to update its projections**, 
  *and/or*
- Running selected projections on individual instance with leader election when projections are stored in DB

**Comment by jeremydmiller** [comment]
@Richard87 Take a look a little deeper in our backlog and you'll see stories for exactly what you're describing. Can't promise an exact timeline of *when* it'll be delivered, but I'm hoping to be working on that extensively after May.

**Reply by Richard87** [reply]
Awesome, will look trough it a bit more!
